opm/simulators/wells/TargetCalculator.cpp

```diff
@@ -93,11 +93,8 @@ RateType TargetCalculator::calcModeRateFromRates(const RateType* rates) const
         return rates[opos] + rates[wpos];
     }
     case Group::ProductionCMode::RESV: {
-        assert(pu_.phase_used[BlackoilPhases::Liquid]);
-        assert(pu_.phase_used[BlackoilPhases::Aqua]);
-        assert(pu_.phase_used[BlackoilPhases::Vapour]);
-        ElemType mode_rate = zero<ElemType>();
-        for (int phase = 0; phase < pu_.num_phases; ++phase) {
+        ElemType mode_rate = rates[0] * resv_coeff_[0];
```

Comment:
The  `zero<ElemType>()` does not seem to have the same number of derivatives as rates. I agree it is strange since the `zero` seems to do the right thing. I can give some more details about the error message I get.

Reply:
The size of `mode_rate =  zero<ElemType>()` is zero 
while `rates[0]` has size 8. 

I suggest we remove `zero` and `ElemType ` and that we go for the simpler solution I  proposed.

Reply:
I do not understand what fails. But your solution is fine.